src/main/kotlin/com/sourcegraph/cody/config/CodyAuthenticationManager.kt

```diff
@@ -129,14 +183,15 @@ class CodyAuthenticationManager(val project: Project) : Disposable {
     else null
   }
 
-  fun setActiveAccount(account: CodyAccount?) {
+  fun setActiveAccount(account: CodyAccount?, previousToken: String?) {
```

Comment:
Can you explain wht that change was necessary?
It looks a bit backwards to me, as one could expect that `CodyAuthenticationManager` should know about previous token instead of being told explicitly what it was.
I see that e.g. you pass `null` there for the initial set, but I do not see why it would be needed vs the old code.

Reply:
`CodyAccount` does not store the token value. `CodyAccount` is only a key. The credentials (the token) is already updated when we call `setActiveAccount`. `com.sourcegraph.cody.config.CodyAccountManager` is responsible for the token storage.